### ISSUE TITLE
NEON code does not build on armv7

### ISSUE THREAD

**Issue by clausecker** [mosra/corrade#171]
The NEON code in your project appears to use intrinsics that only exist on AArch64.  However, you merely check for the presence of NEON extensions, also trying to build this code on AArch32 (aka arm, armv7).  Please check this and either remove the intrinsics in question or change the code so it is only build on AArch64.  Here are the offending intrinsics I found in my first pass (both in `StringView.cpp`):

    vshrn_high_n_u16
    vaddvq_u8

**Comment by mosra** [comment]
Should be fixed in 0169e212f124e678aac1980e3c825f5431d298a5, sorry it took such a massive amount of time.

I also grabbed your patches from the commit referenced above and pushed them as d99afc1b374cd13e00113de238439583531d7804 and be614b914a1724a2426a2703e9dd4afa89c5de08 -- thank you for these! I attempted to do some blind polishing but currently have no automated way to check that they work, so hopefully I didn't break anything. Let me know if I did, I'll try to be more responsive this time.

(Tangential, ideally I'd have some sort of a FreeBSD CI build (and a Big-Endian build, and something where I can test 32-bit ARM...), but CircleCI doesn't support those, and my attempts to build those in a VM failed spectacularly: #166)
